exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ScanBatch.java

```diff
@@ -242,12 +242,12 @@ private IterOutcome internalNext() {
       if (implicitValues != null) {
         String projectMetadataColumn = context.getOptions().getOption(ExecConstants.IMPLICIT_PROJECT_METADATA_COLUMN_LABEL_VALIDATOR);
         if (recordCount > 0) {
-          // Sets the implicit value to false to signal that some results were returned and there is no need for creating an additional record.
-          implicitValues.replace(projectMetadataColumn, Boolean.FALSE.toString());
-        } else if (Boolean.parseBoolean(implicitValues.get(projectMetadataColumn))) {
+          // Sets the implicit value to null to signal that some results were returned and there is no need for creating an additional record.
+          implicitValues.replace(projectMetadataColumn, null);
+        } else if (Boolean.FALSE.toString().equals(implicitValues.get(projectMetadataColumn))) {
```

Comment:
Any reason for use to be so specific here? The rule we've used in column properties is that `true` is if the value, converted to lower case, is "true", `false` otherwise. Else, we end up with values which are neither `true` nor `false` because they are some value like "FALSE".

Maybe we need a utility conversion function: `isTrue(String value)` and `isFalse(String value)` where an invariant is that, for any string `s`, `isTrue(s) = !isFalse(s)`.

Reply:
Values here are written using either using `Boolean.FALSE.toString()` or `Boolean.TRUE.toString()` or unset.